src/odemis/gui/cont/tabs.py

```diff
@@ -7202,6 +7279,139 @@ def _on_autofocus_done(self, future):
         # Enable manual focus when done running autofocus
         self.panel.btn_manual_focus.Enable(True)
 
+    @call_in_wx_main
+    def _on_specswitch_align(self, event):
+        """
+        Called when one of the retract and engage spec-switch mirror buttons are pressed.
+        It will start movement for the spec-switch mirror (retract/engage).
+        GUI is adjusted in this method and therefore make the call in main.
+        """
+        main = self.tab_data_model.main
+        spec_switch = main.spec_switch.getMetadata()
+        # assign the button which got the click event
+        btn = event.GetEventObject()
+
+        # GUI stream bar controller pauses the stream
+        self._stream_controller.pauseStreams()
```

Comment:
The stream should not always be paused. I think it would be fine to just not do anything (leave the stream play).
If you want to be fancy, you could pause it when the "engage" button is pressed (because the user won't see anything), and when the "retract" button is pressed, the stream should play again (because the user will be able to see again image). This is done with `self._ccd_stream.should_update.value = True`.